mypy/types.py

```diff
@@ -1486,112 +1486,112 @@ def keywords_str(self, a: Iterable[Tuple[str, Type]]) -> str:
         ])
 
 
-# These constants define the method used by TypeQuery to combine multiple
-# query results, e.g. for tuple types. The strategy is not used for empty
-# result lists; in that case the default value takes precedence.
-ANY_TYPE_STRATEGY = 0   # Return True if any of the results are True.
-ALL_TYPES_STRATEGY = 1  # Return True if all of the results are True.
+# Combination strategies for boolean type queries
+def ANY_TYPE_STRATEGY(current: bool, accumulated: bool) -> bool:
+    """True if any type's result is True"""
+    if accumulated:
+        raise ShortCircuitQuery()
+    return current
```

Comment:
Maybe replace raising exception with returning a tuple of (current, short_circuit), and break in query_types loop when should_break is true? I don't think using exceptions to return a value is bad if it's a relatively less frequent condition, but in this case it will happen very often. Of course, in this case, all strategies would need to support this interface, so if you think it's too much, it's fine.